src/main/java/co/cask/gcp/gcs/source/AbstractFileBatchSource.java

```diff
@@ -82,15 +81,17 @@ public void prepareRun(BatchSourceContext context) throws Exception {
       conf.set(entry.getKey(), entry.getValue());
     }
 
-    conf.set(INPUT_REGEX_CONFIG, config.fileRegex);
+    if (config.fileRegex != null) {
+      conf.set(INPUT_REGEX_CONFIG, config.fileRegex);
+    }
 
-    FileInputFormat.setInputDirRecursive(job, config.recursive);
+    FileInputFormat.setInputDirRecursive(job, Boolean.TRUE.equals(config.recursive));
```

Comment:
previous comment was about adding a isRecursive(), shouldIgnoreNonExistingFolders() and useFilenameOnly() methods to the config class that return boolean and not Boolean. Those properties should not be public.

Reply:
Oh I thought that the intention was not have null checks hanging around. Will add the getters